server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/KEYSLOT.java

```diff
@@ -38,7 +39,16 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, Channel
       KeyPartitioner partitioner = SecurityActions.getCacheComponentRegistry(respCache)
             .getComponent(KeyPartitioner.class);
 
-      CompletionStage<Integer> cs = CompletableFuture.completedFuture(partitioner.getSegment(arguments.get(1)));
+      CompletionStage<Integer> cs;
+      if (partitioner instanceof HashFunctionPartitioner) {
+         HashFunctionPartitioner hashPartitioner = (HashFunctionPartitioner) partitioner;
+         byte[] key = arguments.get(1);
+         int h = hashPartitioner.getHashForKey(key);
+         cs = CompletableFuture.completedFuture(handler.respServer().segmentSlotRelation().hashToSlot(h));
+      } else {
+         cs = CompletableFuture.completedFuture(partitioner.getSegment(arguments.get(1)));
```

Comment:
This is no longer valid, we should probably just throw an exception maybe?

Reply:
Updated. Now returns an error to the client and returns the current handler CF.